nextcord/embeds.py

```diff
@@ -373,15 +382,25 @@ def set_footer(
             The footer text.
         icon_url: :class:`str`
             The URL of the footer icon. Only HTTP(S) is supported.
-        """
+        icon_file: Optional[:class:`File`]
+            A file to use for the image.
 
+            .. versionadded:: 2.1
+        """
         self._footer = {}
         if text is not EmptyEmbed:
             self._footer["text"] = str(text)
 
+        if icon_url is not EmptyEmbed and icon_file is not EmptyEmbed:
+            raise InvalidArgument("Cannot pass both icon_url and icon_file")
+
         if icon_url is not EmptyEmbed:
             self._footer["icon_url"] = str(icon_url)
 
+        if isinstance(icon_file, File):
+            self._local_files["footer"] = icon_file
```

Comment:
footer image file can be removed from local files if icon url is set afterwards similar to other methods?